core/client/src/main/java/alluxio/client/file/UnderFileSystemFileInStream.java

```diff
@@ -164,7 +164,10 @@ private ByteBuffer allocateBuffer() {
   private void bufferedRead(int len) throws IOException {
     mBuffer.clear();
     int bytesRead = directRead(mBuffer.array(), 0, len);
```

Comment:
(my 2 cents, feel free to decline) : if we make `drectRead` to take `ByteBuffer` instead of `byte[]` as the parameter for dst, then we don't need to deal with `array()`, which might look safer. Meanwhile, the other reference to `directRead` (line 118) may need to be wrapped from `byte[]` to `ByteBuffer`